Celigo.ServiceManager.NetSuite.REST/src/RestClient.cs

```diff
@@ -0,0 +1,203 @@
+            foreach (var param in queryParams.AllKeys)
+            {
+                signingParams.Add(param, queryParams[param]);
+            }
+        }
+
+        protected virtual string ComputeNonce()
+        {
+            byte[] data = new byte[20];
+            using (RNGCryptoServiceProvider rng = new RNGCryptoServiceProvider())
+            {
+                rng.GetBytes(data);
+            }
+
+            int value = Math.Abs(BitConverter.ToInt32(data, 0));
+            return value.ToString();
+        }
+
+        protected virtual string ComputeSignature(string baseString, string tokenSecret)
+        {
+            string key = string.Concat(ConsumerSecret, "&", tokenSecret);
+
+            var encoding = Encoding.ASCII;
+            byte[] keyBytes = encoding.GetBytes(key);
+            byte[] baseStringBytes = encoding.GetBytes(baseString);
+            
+            using (var hmacSha256 = new HMACSHA256(keyBytes))
+            {
+                byte[] baseStringHash = hmacSha256.ComputeHash(baseStringBytes);
+                return Convert.ToBase64String(baseStringHash);
+            }
+        }
+
+        protected virtual long ComputeTimestamp() =>
+            ((long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds);
+
+        protected static StringContent CreateJsonMessageContent<T>(T content) =>
+            new StringContent(JsonSerializer.Serialize(content, SerializerSettings), Encoding.UTF8, "application/json");
+        
+        protected string GetAuthorizationHeaderValue(string account,
+                                                     Uri requestUri,
+                                                     string token,
+                                                     string tokenSecret,
+                                                     string httpMethod)
+        {
+            var oauthParams = this.GetCommonOAuthParameters();
+            this.AddQueryParamsFrom(requestUri, oauthParams);
+            
+            string basePath = requestUri.GetLeftPart(UriPartial.Path);
+            return this.GetAuthorizationHeaderValue(account, basePath, oauthParams, token, tokenSecret, httpMethod);
+        }
+        
+        protected string GetAuthorizationHeaderValue(string account,
+                                                    string baseUrlPath,
+                                                    SortedDictionary<string, string> oauthParams,
+                                                    string token,
+                                                    string tokenSecret,
+                                                    string httpMethod)
+        {
+            oauthParams.Add("oauth_token", token);
+            
+            string E(string data) => Uri.EscapeDataString(data);
+
+            var headerBuilder = new StringBuilder(300);
+            var baseStringBuilder = new StringBuilder(300);
```

Comment:
Is this capacity of 300 arbitrary ?

Reply:
Nope. I kind of sampled a few headers and they varied around 270-280 characters. So, I left a buffer and set at 300.